game/ui/base/transactionscreen.cpp

```diff
@@ -386,18 +386,27 @@ void TransactionScreen::populateControlsAlien()
 {
 	int leftIndex = getLeftIndex();
 	int rightIndex = getRightIndex();
+
 	for (auto &ae : state->agent_equipment)
 	{
+		auto &alienTypeName = ae.first;
+
 		if (!ae.second->bioStorage)
 		{
 			continue;
 		}
 		// Add alien
 		for (auto &b : state->player_bases)
 		{
-			// Removing brainsucker pod from alien containment list
+			auto alienTypeExistingControl = findControlById(type, alienTypeName);
+
 			if (b.second->inventoryBioEquipment[ae.first] > 0 &&
-			    ae.first != "AEQUIPMENTTYPE_BRAINSUCKER_POD")
+
+			    // Removing brainsucker pod from alien containment list
```

Comment:
I don't really like having a complex enough conditional so that comments are needed in the middle - can make a little hard to read.

Maybe this can be separated out so each is evaluated separately? Might need to reverse the comparisons?



e.g. something like:
```
if (b.second->inventoryBioEquipment[ae.first] == 0)
 continue;

//Skip brainsuckers
if (alienTypeName == "AEQUIPMENTYPE_BRAINSUCKER_POD")
  continue;

// Skip if it's already added
if (alienTypeExistingControl)
  continue;

// Create the control
auto control = ... etc etc etc
```

Does that make the logic clearer?

Reply:
Though some things on this list may be removed by #1381 anyway :P

Reply:
I reworked the condition to avoid having so many nested conditions. I also removed the outdated "Brainsucker Pod" reference:

https://github.com/OpenApoc/OpenApoc/blob/12f960b12bf8b7d40fd104f027fedb298ccdcd30/game/ui/base/transactionscreen.cpp#L399-L416